### PR TITLE
Fishchimp/features/fetch unavailability testing

### PR DESCRIPTION
**Pull request by fishchimp**
## Describe your changes
- Pulled changes from John's branch
- Made unit tests for the unavailability fetch (get) method.

#### **Test 1: Success Retrieval of Unavailability Records**
Objective: Verify that the system can successfully retrieve unavailability records for a volunteer with existing records.
Criteria:
- The response status code must be 200, indicating a successful request.
- The response must contain an 'unavailability' key, showing that unavailability data is being returned.
- The 'unavailability' key must be associated with a list, indicating one or more unavailability records are present.

#### **Test 2: No Unavailability Records Found**
Objective: Confirm the system's response when a valid volunteer has no unavailability records.
Criteria:
- The response status code should be 400, indicating a client-side error due to the absence of records.
- To inform the client about the lack of records, the response body must explicitly indicate failure, with a structure matching {'userID': user_id, 'success': False}.

#### **Test  3: Invalid Volunteer User ID**
Objective: Ensure the system correctly handles requests for non-existent users.
Criteria:
The response status code should be 404, indicating that the requested resource (user or unavailability record) was not found.
The response should include a message indicating that the user was not found, making the error clear to the client.

#### **Test 4: Unauthorized Access Attempt**
Objective: Test the system's response to an unauthorized request, simulating a scenario where the requestor is not authenticated.
Criteria:
- The response status code must be 401, signaling unauthorized access due to lack of or invalid authentication.
- The response must contain an error message, providing feedback that authentication is required.

#### **Test 5: Forbidden Access Attempt**
Objective: Validate the system's handling of requests from authenticated users who do not have permission to access the requested data.
Criteria:
- The response status code should be 403, indicating that the requestor is authenticated but lacks the necessary permissions to access the requested resource.
- The response must include an error message, indicating the requestor's insufficient permissions.

## Issue ticket number and link
[Issue FE-146](https://fireapp-emergiq-2024.atlassian.net/browse/FE-146)

NOTE: I'm a bit unsure how to do the tests for test 4 and test 5 as the original method has the @requires_auth decorative

### CODE REVIEW

File: controllers/v2/unavailability/api.py [hunk 1]
```diff
@@ -59,8 +61,8 @@ def get(self, user_id):
             else:
                 return jsonify({'userID': user_id, 'success': False}), 400
 
-    @requires_auth
```
Comment (BenMMcLean) [inline controllers/v2/unavailability/api.py:62]: Do not remove these. Make a global variable that allows you to disable these checks.

File: services/optimiser/input_processing.py [hunk 2]
```diff
@@ -349,7 +349,7 @@ def get_position_qualification(session, position_id):
     return qualification_id
 
 
-# This can be called by api from postman, to test easily
+# This can be called by api from postman, to tests easily
```
Comment (BenMMcLean) [inline services/optimiser/input_processing.py:352]: Please move this into the test directory. I know you didn't write it, but I would appreciate it.

**Review by BenMMcLean — CHANGES_REQUESTED** [review]
Write a way to disable authentication checking globally, do not remove auth annotations.

**Review by BenMMcLean — CHANGES_REQUESTED** [review]
Why are pushes from main showing up in your diff? Please merge main normally and I will review your work

**Comment by fishchimp** [comment]
> Why are pushes from main showing up in your diff? Please merge main normally and I will review your work

Hi Ben, apologies, I have a habit of pulling from the main branch to merge incoming changes. I'll make sure to only do merges without pulling from now on. I can't merge from main anymore as the branch is already up to date

**Comment by BenMMcLean** [comment]
I understand, but I can't merge or review your branch in its current state as GitHub is reporting conflicts and the pipelines aren't running. Besides, it is difficult for me to assess your changes when everyone else's are mixed in as I'm not sure what exactly you did. Please roll back to before you pulled main and do a merge commit from main.

**Comment by fishchimp** [comment]
Should I delete this branch and create a new one? I think that would make it a lot easier

**Comment by fishchimp** [comment]
Should I delete this PR and redo the process on a new branch? That would make it easier.

**Comment by BenMMcLean** [comment]
I would prefer if you didn't, but I don't mind

**Comment by BenMMcLean** [comment]
@fishchimp Please resolve the issues on this branch. This PR has been open for a week and I will not merge your other work until this one is complete

**Comment by fishchimp** [comment]
Hi @BenMMcLean , I'm currently trying to solve the merge conflicts and am still running into some git merging and rebasing issues. In the meantime, the features I've implemented here and on my other branch are the same. I would appreciate it if you could take a look at that one. I'll keep trying to solve this merge conflict issue and meet up with John.

**Comment by fishchimp** [comment]
Hi @BenMMcLean,

This Git issue is getting too complicated and taking too much time. Would you agree that we close this PR? My other PR has the same changes without all other people's commits junked in.

**Comment by BenMMcLean** [comment]
yep that's fine

**Pull request closed without merging**